class-auth.php

```diff
@@ -102,8 +102,28 @@ public function register_rest_routes() {
 	public function add_cors_support() {
 		$enable_cors = defined( 'JWT_AUTH_CORS_ENABLE' ) ? JWT_AUTH_CORS_ENABLE : false;
 
-		if ( $enable_cors && !headers_sent() ) {
-			$headers = apply_filters( 'jwt_auth_cors_allow_headers', 'X-Requested-With, Content-Type, Accept, Origin, Authorization, Cookie' );
+		global $wp_version;
+		$parsedVersion = array_map( function ( $value ) {
+			return intval( $value );
+		}, explode( '.', $wp_version ) );
+
+		$major = $parsedVersion[0] ?? 0;
+		$minor = $parsedVersion[1] ?? 0;
+
+		// Hook exists since 5.5.0
+		$versionCheck = $major > 5 || ($major === 5 && $minor >= 5 );
+
+		if ( $enable_cors && $versionCheck ) {
```

Comment:
Seems like the `$enable_cors` condition should be separated. Since the function does not do anything else, you can negate the condition and turn it into an early return if not enabled.

Reply:
Thanks, didn't know about that one!